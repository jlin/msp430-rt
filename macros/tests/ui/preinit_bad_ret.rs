#![no_main]

use msp430_rt_macros::{entry, pre_init};

#[entry]
fn bar() -> ! {
    loop {}
}

#[pre_init]
unsafe fn foo() -> u32 {
    3
}
