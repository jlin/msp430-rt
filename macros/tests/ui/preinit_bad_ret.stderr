error: `#[pre_init]` function must have signature `unsafe fn()`
  --> tests/ui/preinit_bad_ret.rs:11:1
   |
11 | unsafe fn foo() -> u32 {
   | ^^^^^^^^^^^^^^^^^^^^^^
